sound/soc/sof/intel/hda-dai.c

```diff
@@ -422,56 +422,176 @@ static struct snd_soc_cdai_ops sof_probe_compr_ops = {
 struct snd_soc_dai_driver skl_dai[] = {
 {
 	.name = "SSP0 Pin",
+	.playback = {
+		.stream_name = "ssp0 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
+	.capture = {
+		.stream_name = "ssp0 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "SSP1 Pin",
+	.playback = {
+		.stream_name = "ssp1 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
+	.capture = {
+		.stream_name = "ssp1 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "SSP2 Pin",
+	.playback = {
+		.stream_name = "ssp2 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
+	.capture = {
+		.stream_name = "ssp2 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "SSP3 Pin",
+	.playback = {
+		.stream_name = "ssp3 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
+	.capture = {
+		.stream_name = "ssp3 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "SSP4 Pin",
+	.playback = {
+		.stream_name = "ssp4 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
+	.capture = {
+		.stream_name = "ssp4 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "SSP5 Pin",
+	.playback = {
+		.stream_name = "ssp5 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
+	.capture = {
+		.stream_name = "ssp5 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "DMIC01 Pin",
+	.capture = {
+		.stream_name = "DMIC01 Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 {
 	.name = "DMIC16k Pin",
+	.capture = {
+		.stream_name = "DMIC16k Rx",
+		.channels_min = 1,
+		.channels_max = 8,
+	},
 },
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 {
 	.name = "iDisp1 Pin",
 	.ops = &hda_link_dai_ops,
+	.playback = {
+		.stream_name = "iDisp1 Tx",
+		.channels_min = 1,
+		.channels_max = 8,
```

Comment:
this needs to be 4 for a DMIC but we can fix this in a follow-up patch, no reason to gate integration